### PR TITLE
release notes

### PR DESCRIPTION
**Pull request by ckaadic**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
In December the currently used alerts endpoint was moved to legacy and a new endpoint (v2) was released (see [List alerts](https://learn.microsoft.com/en-us/graph/api/security-list-alerts_v2?view=graph-rest-1.0&tabs=http) and [Get Alert](https://learn.microsoft.com/en-us/graph/api/security-alert-get?view=graph-rest-1.0&tabs=http)). I marked the current commands for get-alert-details and search-alerts as deprecated and added new commands for v2. The changes do not break backward compatibility, but the v2 endpoint does not provide exactly the same information as the legacy one.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ckaadic** [comment]
Will be adding the documentation to the readme next week.

**Comment by adi88d** [comment]
@ckaadic Please fille the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form

**Comment by ckaadic** [comment]
@adi added the documentation to readme, fixed the lint/validate feedback

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 1]
```diff
@@ -403,6 +453,58 @@ def get_alert_details_command(client: MsGraphClient, args):
     return hr, ec, alert_details
 
 
+def get_alert_details_v2_command(client: MsGraphClient, args):
+    alert_id = args.get('alert_id')
+    alerts_v2 = True
+
+    alert_details = client.get_alert_details(alert_id, alerts_v2)
```
Comment (adi88d) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:460]:
```suggestion


    alert_details = client.get_alert_details(alert_id, True)
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 2]
```diff
@@ -403,6 +453,58 @@ def get_alert_details_command(client: MsGraphClient, args):
...
+
+    hr = '## Microsoft Security Graph Alert Details - {}\n'.format(alert_id)
+
+    basic_properties_title = 'Basic Properties'
+    basic_properties = {
+        'ActorDisplayName': alert_details['actorDisplayName'],
```
Comment (adi88d) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:466]: please use `get` for the other properties too and the context
```suggestion
        'ActorDisplayName': alert_details.get('actorDisplayName'),
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 3]
```diff
@@ -403,6 +453,58 @@ def get_alert_details_command(client: MsGraphClient, args):
...
+        'Title': alert_details['title']
+    }
+    hr += tableToMarkdown(basic_properties_title, basic_properties, removeNull=True)
+
+    if alert_details['comments']:
+        comments = alert_details['comments']
+        if comments:
+            comments_hr = '### Comments provided for Alert\n'
+            for comment in comments:
+                comments_hr += '- {}\n'.format(comment['comment'])
```
Comment (adi88d) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:488]:
```suggestion
            for comment in comments:
                if comment.get('comment'):
                    comments_hr += '- {}\n'.format(comment['comment'])
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 4]
```diff
@@ -60,8 +60,27 @@ def search_alerts(self, last_modified, severity, category, vendor, time_from, ti
         response = self.ms_client.http_request(method='GET', url_suffix=cmd_url, params=params)
         return response
 
-    def get_alert_details(self, alert_id):
-        cmd_url = f'security/alerts/{alert_id}'
+    def search_alerts_v2(self, last_modified, severity, time_from, time_to, filter_query):
```
Comment (adi88d) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:63]: `search_alerts_v2` looks similar to `search_alerts` please try to change the `search_alerts` method by adding the parameter `use_alerts_v2=False` so it works for v1 and v2

**Review by adi88d — CHANGES_REQUESTED** [review]
Hi @ckaadic and thank you for your contribution! it looks great!
- please add command examples for `msg-search-alerts-v2` and `msg-get-alert-details-v2`
- no need to deprecate the old commands, please revert

see my other comments below

**Comment by ckaadic** [comment]
Hi @adi88d, validate wanted me to push image version again. Followed all your comments.

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py [hunk 5]
```diff
@@ -189,6 +189,37 @@ def search_alerts_command(client: MsGraphClient, args):
...
+    last_modified = args.get('last_modified')
+    severity = args.get('severity')
+    time_from = args.get('time_from')
+    time_to = args.get('time_to')
+    filter_query = args.get('filter')
+    alerts = client.search_alerts(last_modified, severity, None, None, time_from, time_to, filter_query, True)['value']
+    outputs = []
+    for alert in alerts:
+        outputs.append({
+            'ID': alert['id'],
```
Comment (adi88d) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py:202]:
```suggestion
            'ID': alert.get('id'),
```

File: Packs/MicrosoftGraphSecurity/pack_metadata.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
     "name": "Microsoft Graph Security",
     "description": "Unified gateway to security insights - all from a unified Microsoft Graph\n  Security API.",
     "support": "xsoar",
-    "currentVersion": "2.1.12",
+    "currentVersion": "2.2.0",
```
Comment (adi88d) [inline Packs/MicrosoftGraphSecurity/pack_metadata.json:5]: please bump the version to 2.1.13
and change the `Packs/MicrosoftGraphSecurity/ReleaseNotes/2_2_0.md` file name to `Packs/MicrosoftGraphSecurity/ReleaseNotes/2_1_13.md`
```suggestion
    "currentVersion": "2.1.13",
```

**Review by adi88d — CHANGES_REQUESTED** [review]
great job!

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 7]
```diff
@@ -238,6 +238,81 @@ script:
     - contextPath: MsGraph.Alert.Provider
       description: Alert provider
       type: string
+  - arguments:
+    - default: false
+      description: When the alert was last modified in the following string format - YYYY-MM-DD
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:243]:
```suggestion
      description: The date the alert was last modified in the following string format - YYYY-MM-DD
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 8]
```diff
@@ -238,6 +238,81 @@ script:
...
+      secret: false
+      auto: PREDEFINED
+      predefined:
+      - unknown
+      - informational
+      - low
+      - medium
+      - high
+    - default: false
+      description: The start time (creation time of alert) for the search in the following string format - YYYY-MM-DD
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:267]:
```suggestion
      description: The start time (creation time of the alert) for the search in the following string format - YYYY-MM-DD.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 9]
```diff
@@ -238,6 +238,81 @@ script:
...
+      isArray: false
+      name: time_from
+      required: false
+      secret: false
+    - default: false
+      description: The end time (creation time of alert) for the search in the following string format -  YYYY-MM-DD
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:273]:
```suggestion
      description: The end time (creation time of the alert) for the search in the following string format -  YYYY-MM-DD.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 10]
```diff
@@ -238,6 +238,81 @@ script:
...
+      isArray: false
+      name: time_to
+      required: false
+      secret: false
+    - default: false
+      description: Use this field to filter on any of the alert properties in the format "{property} eq '{property-value}'", e.g. "assignedTo eq 'user1'"
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:279]:
```suggestion
      description: Filter on any of the alert properties in the format "{property} eq '{property-value}'", e.g., "assignedTo eq 'user1'".
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 11]
```diff
@@ -238,6 +238,81 @@ script:
...
+      name: filter
+      required: false
+      secret: false
+    deprecated: false
+    description: List alerts (security issues) within a customer's tenant that Microsoft or partner security solutions have identified.
+    execution: false
+    name: msg-search-alerts-v2
+    outputs:
+    - contextPath: MsGraph.Alert.ID
+      description: Alert ID
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:290]:
```suggestion
      description: Alert ID.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 12]
```diff
@@ -238,6 +238,81 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.Title
+      description: Alert title
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:293]:
```suggestion
      description: Alert title.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 13]
```diff
@@ -238,6 +238,81 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.Category
+      description: Alert category
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:296]:
```suggestion
      description: Alert category.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 14]
```diff
@@ -238,6 +238,81 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.Severity
+      description: Alert severity
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:299]:
```suggestion
      description: Alert severity.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 15]
```diff
@@ -238,6 +238,81 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.CreatedDate
+      description: Alert created date
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:302]:
```suggestion
      description: Alert created date.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 16]
```diff
@@ -238,6 +238,81 @@ script:
...
+      type: date
+    - contextPath: MsGraph.Alert.FirstActivityDateTime
+      description: Alert first activity time
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:305]:
```suggestion
      description: Alert first activity time.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 17]
```diff
@@ -238,6 +238,81 @@ script:
...
+      type: date
+    - contextPath: MsGraph.Alert.LastActivityDateTime
+      description: Alert last activity time
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:308]:
```suggestion
      description: Alert last activity time.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 18]
```diff
@@ -238,6 +238,81 @@ script:
...
+      type: date
+    - contextPath: MsGraph.Alert.Status
+      description: Alert status
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:311]:
```suggestion
      description: Alert status.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 19]
```diff
@@ -238,6 +238,81 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.Provider
+      description: Alert provider
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:314]:
```suggestion
      description: Alert provider.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 20]
```diff
@@ -284,6 +359,45 @@ script:
     - contextPath: MsGraph.Alert.Provider
       description: Alert provider
       type: Unknown
+  - deprecated: false
+    description: Get details for a specific alert.
+    execution: false
+    name: msg-get-alert-details-v2
+    outputs:
+    - contextPath: MsGraph.Alert.ID
+      description: Alert ID
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:368]:
```suggestion
      description: Alert ID.
```

Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:368]:
```suggestion
      description: Alert ID.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 21]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.Title
+      description: Alert title
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:371]:
```suggestion
      description: Alert title.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 22]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.Category
+      description: Alert category
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:374]:
```suggestion
      description: Alert category.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 23]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.Severity
+      description: Alert severity
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:377]:
```suggestion
      description: Alert severity.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 24]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.CreatedDate
+      description: Alert created date
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:380]:
```suggestion
      description: Alert created date.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 25]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: date
+    - contextPath: MsGraph.Alert.FirstActivityDateTime
+      description: Alert first associated activity time
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:383]:
```suggestion
      description: Alert first associated activity time.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 26]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: date
+    - contextPath: MsGraph.Alert.LastActivityDateTime
+      description: Alert latest associated activity time
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:386]:
```suggestion
      description: Alert latest associated activity time.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 27]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: date
+    - contextPath: MsGraph.Alert.Status
+      description: Alert status
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:389]:
```suggestion
      description: Alert status.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 28]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: string
+    - contextPath: MsGraph.Alert.Provider
+      description: Alert provider
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:392]:
```suggestion
      description: Alert provider.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml [hunk 29]
```diff
@@ -284,6 +359,45 @@ script:
...
+      type: Unknown
+    arguments:
+    - default: true
+      description: The Alert ID - Provider-generated GUID/unique identifier.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.yml:396]:
```suggestion
      description: The alert ID. Provider-generated GUID/unique identifier.
```

File: Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/README.md [hunk 30]
```diff
@@ -138,3 +138,111 @@ Update an editable alert property within any integrated solution to keep alert s
...
+
+#### Base Command
+
+`msg-search-alerts-v2`
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| last_modified | When the alert was last modified in the following string format - YYYY-MM-DD. Possible values are: Last24Hours, Last48Hours, LastWeek. | Optional | 
+| severity | Alert severity - set by vendor/provider. Possible values are: unknown, informational, low, medium, high. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/README.md:155]:
```suggestion
| severity | Alert severity set by the vendor/provider. Possible values are: unknown, informational, low, medium, high. | Optional | 
```

**Comment by ShirleyDenkberg** [comment]
@adi88d Doc review completed.

**Comment by adi88d** [comment]
Hello,
Your contribution is much appreciated. However, this endpoint it is not consistent with the API of this integration, and we will open an internal ticket to add it to the `Microsoft Graph Security` integration.
thanks again

**Comment by ckaadic** [comment]
Hi @adi88d is there a way for me to track the progress?

**Comment by ShacharKidor** [comment]
Hi @ckaadic, 
I really apologize for the delay in response. 
@adi88d will contact you soon with explanations. 
Please reach out to me if you'll have more questions or need anything - available on Slack (DFIR). 

FYI - @Mark-Hillenaar.

**Pull request closed without merging**